modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/adapter/DocumentAdapterService.java

```diff
@@ -40,62 +39,37 @@
     public static final ComponentName NAME = new ComponentName(ComponentName.DEFAULT_TYPE,
             "org.nuxeo.ecm.core.api.DocumentAdapterService");
 
-    private static final Log log = LogFactory.getLog(DocumentAdapterService.class);
+    protected static final String XP = "adapters";
 
     /**
      * Document adapters
      */
     protected Map<Class<?>, DocumentAdapterDescriptor> adapters;
 
-    public DocumentAdapterDescriptor getAdapterDescriptor(Class<?> itf) {
-        return adapters.get(itf);
-    }
-
-    /**
-     * @since 5.7
-     */
-    public DocumentAdapterDescriptor[] getAdapterDescriptors() {
-        Collection<DocumentAdapterDescriptor> values = adapters.values();
-        return values.toArray(new DocumentAdapterDescriptor[values.size()]);
-    }
-
-    public void registerAdapterFactory(DocumentAdapterDescriptor dae) {
-        adapters.put(dae.getInterface(), dae);
-        log.info("Registered document adapter factory " + dae);
-    }
-
-    public void unregisterAdapterFactory(Class<?> itf) {
-        DocumentAdapterDescriptor dae = adapters.remove(itf);
-        if (dae != null) {
-            log.info("Unregistered document adapter factory: " + dae);
-        }
-    }
-
     @Override
-    public void activate(ComponentContext context) {
+    public void start(ComponentContext context) {
         adapters = new ConcurrentHashMap<>();
+        adapters.putAll(
+                this.<DocumentAdapterDescriptor> getRegistryContributions(XP)
+                    .stream()
+                    .collect(Collectors.toConcurrentMap(DocumentAdapterDescriptor::getInterface, Function.identity())));
```

Comment:
Why not just assign directly, no need for `putAll`. Also if nobody changes the map after creation the concurrent aspect isn't needed.